index.js

```diff
@@ -8,15 +8,20 @@ function createRavenMiddleware(Raven, options = {}) {
     actionTransformer = identity,
     stateTransformer = identity,
     breadcrumbCategory = "redux-action",
-    filterBreadcrumbActions = filter
+    filterBreadcrumbActions = filter,
+    getUserContext
   } = options;
 
   return store => {
     let lastAction;
 
     Raven.setDataCallback((data, original) => {
+      const state = store.getState();
       data.extra.lastAction = actionTransformer(lastAction);
-      data.extra.state = stateTransformer(store.getState());
+      data.extra.state = stateTransformer(state);
+      if (getUserContext) {
+        data.user = getUserContext(state);
```

Comment:
Do you think we should add a note that the original state, and not the transformed one?

Reply:
You mean, a note about it overwriting any existing user context?

Reply:
No, but I did indeed add a notice about that.

But, a note saying that the `getUserContext` is executed with the original redux-state, and _not_ the state transformed via the `stateTransformer`.

Reply:
In the docs, as long as we refer to it as “Redux state” I think that’s clear. In the code, I think it’s self explanatory.